### PR TITLE
Add imageCredentials.productCode configuration to support AWS Marketplace repositories

### PR DESCRIPTION
**Pull request by michael-jong-ds**
We currently have decided to use a product code as repository prefix for our marketplace listing.
Screenshot from 2024-03-15 15-33-38

However, in the helm charts, it is hardcoded to datasaur
Screenshot from 2024-03-15 15-32-53

to support this, I am adding a field to values.yaml to replace the hard coded path to whatever prefix is provided.

### CODE REVIEW

File: charts/datasaur/values.yaml [hunk 1]
```diff
@@ -170,6 +170,7 @@ imageCredentials:
   username: ""
   password: ""
   registry: ""
+  productCode: ""
```
Comment (tonoman3g) [inline charts/datasaur/values.yaml]: kenapa termnya namanya productCode? @michael-jong-ds

Comment (michael-jong-ds) [reply charts/datasaur/values.yaml]: bingung tbh namanya mau apa :laughing: 
masa marketplace sku, jadi kyk terekspos

Comment (daniel-jeremia) [reply charts/datasaur/values.yaml]: kasi nama `prefix` aja..

Comment (michael-jong-ds) [reply charts/datasaur/values.yaml]: updated

**Review by daniel-jeremia — APPROVED** [review]

**Comment by michael-jong-ds** [comment]
wait, I found something else, still testing

**Review by tonoman3g — APPROVED** [review]

**Pull request merged**